kato/kato-aws/src/main/groovy/com/netflix/spinnaker/kato/aws/deploy/DefaultLaunchConfigurationBuilder.groovy

```diff
@@ -134,7 +135,7 @@ class DefaultLaunchConfigurationBuilder implements LaunchConfigurationBuilder {
     settings = settings.copyWith(securityGroups: securityGroupIds)
 
     String name = createName(settings)
-    String userData = getUserData(settings.baseName, name, settings.region, settings.account, settings.environment, settings.accountType)
+    String userData = getUserData(settings.baseName, name, settings.region, settings.account, settings.environment, settings.accountType, settings.base64UserData)
```

Comment:
if the settings object has base64UserData on it then the calculation should happen outside this method

the rolling push deploy strategy (we don't recommend it, but we use it), reads the existing launch configuration via `buildSettingsFromLaunchConfiguration`, applies the passed in changes, then calls `buildLaunchConfiguration`. I think as this stands it would end up appending duplicated userdata on each rolling push operation.